sdk/keyvault/Azure.Security.KeyVault.Shared/tests/PremiumOnlyAttribute.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Azure.Security.KeyVault.Tests
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class PremiumOnlyAttribute : NUnitAttribute, IApplyToTest
+    {
+        public void ApplyToTest(Test test)
+        {
+            // Test preparation was previously successfully creating premium SKUs (not available in every cloud), so assume premium.
+            string sku = Environment.GetEnvironmentVariable("KEYVAULT_SKU") ?? "premium";
```

Comment:
This won't work with -OutFile. Please don't hardcode environment variables into more places in test code after we did all this work to reduce the coupling.

Reply:
I forgot. I'll add this to the test resources class. It still needs to be an environment variable from the test-resources.json, though - that's the only way we're going to know if a premium or standard SKU was created (KV falls back to standard if premium can't be created without error).